src/test/java/org/jenkinsci/plugins/workflow/cps/CpsBodyExecutionTest.java

```diff
@@ -206,6 +206,10 @@ public boolean start() throws Exception {
                     "semaphore 'wait'\n", true));
             WorkflowRun b = p.scheduleBuild2(0).waitForStart();
             SemaphoreStep.waitForStart("wait/1", b);
+            String xml = ((CpsFlowExecution) b.getExecution()).programPromise.get().asXml();
+            System.out.print(xml);
+            assertThat(xml, not(containsString(s.getWorkspaceFor(p).getRemote())));
+            assertThat(xml, not(containsString("ExecutorStepExecution")));
```

Comment:
Change `true` to `false` in `ParallelStepExecution` and you will see all sorts of junk here.